analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -63,21 +75,103 @@ public override void ExecutionCompleted()
         var alwaysTrue = trueOperations.Except(falseOperations);
         var alwaysFalse = falseOperations.Except(trueOperations);
 
-        foreach (var operation in alwaysTrue)
+        foreach (var (operation, block) in alwaysTrue.Select(x => (x.Key, x.Value)))
         {
-            ReportIssue(operation, true);
+            ReportIssue(operation, block, true);
         }
-        foreach (var operation in alwaysFalse)
+        foreach (var (operation, block) in alwaysFalse.Select(x => (x.Key, x.Value)))
         {
-            ReportIssue(operation, false);
+            ReportIssue(operation, block, false);
         }
 
         base.ExecutionCompleted();
     }
 
-    private void ReportIssue(IOperation operation, bool conditionEvaluation)
+    private void ReportIssue(IOperation operation, BasicBlock block, bool conditionIsTrue)
```

Comment:
Having "IsTrue" on a `bool` doesn't feel right :)
```suggestion
    private void ReportIssue(IOperation operation, BasicBlock block, bool conditionValue)
```

Reply:
Also below